### PR TITLE
CJose binaries & instructions on updating

### LINKED ISSUE #29: JWK generation &  base64url encoding (cjose)

### LINKED ISSUE #30: JWE decryption (cjose)

### PR DESCRIPTION
**Pull request by sashei**
ECDH generation & JWE decryption
resolves #29
resolves #30

### CODE REVIEW

**Comment by sashei** [comment]
@linuxwolf sorry, this is a massive PR -- to filter things out, you should just be able to look at the .swift ones (interacting with `cjose` & attendant specs), and the `update-dependencies` script.

**Comment by sashei** [comment]
fixes #29 
fixes #30

File: lockbox-ios/FxA/FxAWebView.swift [hunk 1]
```diff
@@ -0,0 +1,25 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import WebKit
+import Foundation
+
+class FxAWebView : WKWebView, WKNavigationDelegate {
+    let oauthBaseURL = "https://scoped-keys-oct10.dev.lcip.org/"
```
Comment (linuxwolf) [inline lockbox-ios/FxA/FxAWebView.swift]: why the separate values here?

Comment (sashei) [reply lockbox-ios/FxA/FxAWebView.swift]: The base URL will presumably change once these are in prod. FxAWebView is only sketched out right now, also -- I can remove it from this commit.

Comment (linuxwolf) [reply lockbox-ios/FxA/FxAWebView.swift]: Ideally, we use [configuration discovery](https://developer.mozilla.org/en-US/docs/Mozilla/Tech/Firefox_Accounts/Introduction#Endpoint_Discovery) to configure the endpoints.  I'd like to do that from the WebExtension, but caching needs to be figured out.  In the meantime, its static configuration is essentially a superset of the configuration info.

That all said, it's not a blocker for me either way.

File: docs/updating-libraries.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+## C Libraries
+
+Lockbox for iOS uses [cjose](https://github.com/cisco/cjose) for JOSE encryption & decryption in the FxA OAuth flow. The binaries to support this external dependency are provided along with this code (in the `lockbox-ios/binaries` folder). However, if you find yourself wanting to update this library (or either of its dependencies, Jansson or OpenSSL), navigate to the `scripts` directory and run `./update-dependencies`
+
+## Datastore
+
+Lockbox for iOS uses the Lockbox [Datastore](https://github.com/mozilla-lockbox/lockbox-datastore) to provide data storage & encryption. The datastore code is bundled into a `bundle.js` file using [browserify](http://browserify.org/) (available via npm) and provided along with this code (in the `lockbox-datastore` folder). However, if you find yourself wanting to update this library, clone the source code, install *its* dependencies with `npm install`, and run `browserify lib/index.js -s DataStoreModule -o bundle.js`. Replace the version at `lockbox-datastore/bundle.js` with the newly browserify-ed `bundle.js`.
```
Comment (joeyg) [inline docs/updating-libraries.md]: Awesome! Thanks for documenting this!

**Comment by sashei** [comment]
@linuxwolf I've updated with a .tar version of the c libraries.

File: docs/updating-libraries.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
```
Comment (linuxwolf) [inline docs/updating-libraries.md]: this is ok for now, but

1. we'll want to find another way to help keep this up-to-date (git submodules?)
2. the `browserify` step ought to be semi-automatable, possibly as part of `update-dependencies`

**Comment by m8ttyB** [comment]
Is QA needed to verify specifics that land with PR?

**Comment by linuxwolf** [comment]
@m8ttyB this patch a developer chore to make it easier (and reproducible) to get a development environment together; I'm not sure how much verification is needed.

**Pull request merged**